blog/2025-07-21-openhab-5-0-release.md

```diff
@@ -0,0 +1,323 @@
+The core runtime handles the lifecycle of them for you, they are automatically created, updated and deleted when the script is reloaded or removed.
+
+We are currently working on implementing this in the JavaScript helper library, which will provide an easy-to-use JavaScript API to those new core features.
+Stay tuned for the next [openhab-js release](https://github.com/openhab/openhab-js) to get started with this new feature from your JavaScript scripts!
+
+## Main UI Enhancements
+
+_Florian Hotze ([@florian-h05](https://github.com/florian-h05)), openHAB Maintainer_
+
+### Sitemap Editor Drag & Drop
+
+When designing sitemaps through the UI, it was very cumbersome to move elements around, as you had to delete and re-add them in the desired order, or cut and paste the DSL code in the code tab.
+With openHAB 5.0, we have introduced a drag and drop feature to the sitemap editor, allowing you to easily rearrange elements in the sitemap:
+
+<p align="center"><img style="max-width: 70%;" src="/uploads/2025-07-21-openhab-5-0-release/sitemap-edit-dragdrop.gif"/></p>
+
+Many thanks to Mark Herwege ([@mherwege](https://github.com/mherwege)) for implementing this feature!
+
+### Model Editor Drag & Drop
+
+Similarly to the sitemap editor, the model editor now supports drag and drop for moving semantic tags around.
+This not only makes it faster to build your semantic model, it also makes it easier to ensure that your model is valid and follows the best practices by introducing numerous validation checks.
+
+<div align="center" class="row">
+  <img style="max-width: 35%; min-width: 250px;" src="/uploads/2025-07-21-openhab-5-0-release/model-dragdrop-1.gif"/>
+  <img style="max-width: 35%; min-width: 250px;" src="/uploads/2025-07-21-openhab-5-0-release/model-dragdrop-2.gif"/>
+</div>
+
+A big thank you to Mark Herwege ([@mherwege](https://github.com/mherwege)) for working on this game-changing feature!
+
+### Regenerate Rules from Templates
+
+### Copy DSL & YAML Definition: Connecting UI with Text Configuration
+
+_Jimmy Tanagra ([@jimtng](https://github.com/jimtng)), openHAB Maintainer_
+
+openHAB 5.0 introduces a convenient option for users who wish to convert their setup into text-based configuration.
+Definitions of **Items** and **Things** can now be copied in **DSL** or the new **YAML** file format, making it easier to reuse, back up, or migrate content into textual configuration files.
+
+This feature is intended purely as a convenience. Users can continue to rely entirely on the UI configuration if they prefer.
+
+**Where you'll find it:**
+
+- **Things list** and **Items list** – Select multiple entries to reveal a `Copy` button
+- **Thing Details** and **Item Details** page – A `Copy File Definition` button is available at the bottom of the page
+- **Thing Inbox** – A `Copy File Definition` button is available when clicking on an entry in the inbox
```

Comment:
Thanks @lolodomo for catching this missing one. I'd like to add a little extra hint here, if that's OK
```suggestion
- **Thing Inbox** – Clicking on an inbox entry reveals a `Copy File Definition` button, allowing you to save the discovered Thing in a file-based configuration rather than importing it into the UI-managed database
```